src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -232,7 +232,11 @@ private static void RemoveIntersectingLocals(Dictionary<LocalSymbol, LocalDefUse
     internal class LocalDefUseInfo
     {
         // stack at variable declaration, may be > 0 in sequences.
-        public readonly int stackAtDeclaration;
+        public int StackAtDeclaration { get; private set; }
+        private readonly ObjectPool<LocalDefUseInfo> _pool;
```

Comment:
made ```LocalDefUseSpan``` a struct
```LocalDefUseInfo``` should be a class though since it owns lazy initialized a d freeable resources.